### PR TITLE
chore: [FX-4009] integrate disable state

### PR DESCRIPTION
**Pull request by dmaklygin**
[FX-4009]

### Description

We need to integrate `disabled` state to Rich Text Editor

### Screenshots

Forms : RichTextEditor - RichTextEditor ⋅ Storybook 🔊 2023-06-08 19-36-03

### Development checks

- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4009]: https://toptal-core.atlassian.net/browse/FX-4009?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by dmaklygin** [comment]
@toptal-anvil ping reviewers

**Review by TomasSlama — APPROVED** [review]
Tested in temploy, works properly

File: packages/picasso/src/RichTextEditor/test.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      id='test-editor'
+      placeholder='placeholder+1'
+      testIds={{ wrapper: 'wrapper-test-id-1' }}
+      {...props}
+    />
+  )
+
+describe('RichTextEditor', () => {
+  beforeEach(() => {
+    mockedLexicalEditor.mockClear()
```
Comment (mkrl) [inline packages/picasso/src/RichTextEditor/test.tsx:35]: Just curious, does it do the same as included in `afterEach` `jest.clearAllMocks()`? If we're doing the cleanup twice, the before cleanup might not be needed

**Review by mkrl — APPROVED** [review]
Looks good, tested on temploy, nice job!

**Pull request merged**